src/App.tsx

```diff
@@ -3,34 +3,63 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App: React.FC = () => {
-  return (
-    <div className="page">
-      <div className="page-content">
-        <div className="box">
-          <div className="field">
-            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-            <label htmlFor="search-query" className="label">
-              Search movie
-            </label>
-
-            <div className="control">
-              <input
-                type="text"
-                id="search-query"
-                className="input"
-                placeholder="Type search word"
-              />
+type State = {
+  query: string
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    query: '',
+  };
+
+  render() {
+    const visibleMovies = moviesFromServer
+      .filter(movie => {
+        const movieTitleCheck = movie.title.toLowerCase()
+          .includes(this.state.query.toLowerCase());
+        const movieDescriptionCheck = movie.description.toLowerCase()
+          .includes(this.state.query.toLowerCase());
+
+        return movieTitleCheck || movieDescriptionCheck;
+      });
```

Comment:
try to avoid such nesting. You can create auxiliary function in which you will accept the text of the movie and compare it with the entered substring. 
and then you code may look `const visibleMovies = moviesFromServer.filter(movie => compareString(movie))`, 
and take this calculation out of the render method to 14 line

![image](https://user-images.githubusercontent.com/70009205/201036672-b78eadf6-b652-4f2a-b283-8b71b9e857ae.png)